### PR TITLE
🪟 🎉  Add temporary dropdown component for geography selection

### PR DESCRIPTION
**Pull request by joeykmh**
## What
Adds a temporary implementation of a dropdown for selecting a data geography. A more nicely designed version is being implemented here: https://github.com/airbytehq/airbyte/pull/18543 but is currently on hold because we may switch away from `react-select`. This PR just introduces a simple component that we can discard or alter later.

image

## How
- Adds a `<DataGeographyDropdown>` component
- Inserts a custom link under `react-select`'s `MenuList` to display a link to a Google form where users can request new geographies

### CODE REVIEW

File: airbyte-webapp/src/components/common/DataGeographyDropdown/DataGeographyDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import { links } from "utils/links";
+
+import styles from "./DataGeographyDropdown.module.scss";
+
+interface DataGeographyDropdownProps {
+  geographies: Geography[];
+  isDisabled?: boolean;
+  onChange: (value: Geography) => void;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  value: any; // our current dropdown types options as any anyway
```
Comment (timroes) [inline airbyte-webapp/src/components/common/DataGeographyDropdown/DataGeographyDropdown.tsx:18]: Could we still type here as `Geography`? So we break the `any` polution with this component?

File: airbyte-webapp/src/packages/cloud/views/workspaces/DataResidencyView/DataResidencyView.tsx [hunk 2]
```diff
@@ -35,6 +35,7 @@ export const DataResidencyView: React.FC = () => {
     values: DefaultDataResidencyFormValues,
     { resetForm }: FormikHelpers<DefaultDataResidencyFormValues>
   ) => {
+    console.log(values);
```
Comment (timroes) [inline airbyte-webapp/src/packages/cloud/views/workspaces/DataResidencyView/DataResidencyView.tsx:38]: should stil be removed.

Comment (josephkmh) [reply airbyte-webapp/src/packages/cloud/views/workspaces/DataResidencyView/DataResidencyView.tsx]: Whoops, thanks!

File: airbyte-webapp/src/components/common/DataGeographyDropdown/DataGeographyDropdown.tsx (hunk 1 above)
Comment (josephkmh) [reply airbyte-webapp/src/components/common/DataGeographyDropdown/DataGeographyDropdown.tsx]: Sure, I just updated the `value` and references to use `Geography`

**Review by timroes — APPROVED** [review]
Code LGTM, tested locally works as expected.

**Pull request merged**